### PR TITLE
Add spell checking

### LINKED ISSUE #43: Spell check does not appear to run as part of the CI
Caught several spelling mistakes in a recent PR despite the CI returning all green.  We probably need to add an explicit github action that runs `make check`.

### PR DESCRIPTION
**Pull request by SteVwonder**
- Add spell checking via `make spelling` and `make check` (copied from the rfc repo).
- Add a GitHub action to run `make spelling` as part of the CI.
- Add requirements.txt for easy dependency installation (used by the github action)

Closes #43

### CODE REVIEW

**Comment by gonsie** [comment]
Those who turn on spelling must fix all the mistakes. Here is the error line:

```
faqs.rst:14:natively:: on GitHub tracking the progress towards the goal of natively compiling on a mac. In the meantime, you can use Docker, see: 
```

Once the tests pass it LGTM!

**Comment by SteVwonder** [comment]
After a few force pushes, it looks like the github action is working and happy now.  Conveniently, there was a word that it error'd out on that my local machine didn't.  So I can at least anecdotally confirm that the check fails if there is a spelling mistake.  Thanks @gonsie for getting that fixed up in the RFC repo.

File: quickstart.rst [hunk 1]
```diff
@@ -182,7 +182,7 @@ A flux session can be also be started under `Slurm <https://github.com/chaos/slu
   srun: job 1136410 has been allocated resources
   $
 
-After broker wireup is completed, the Flux session starts an “initial program” on rank 0 broker. By default, the initial program is an interactive shell, but an alternate program can be supplied on the ``flux start`` command line. Once the initial program terminates, the Flux session is considered complete and brokers exit.
+After broker wire up is completed, the Flux session starts an “initial program” on rank 0 broker. By default, the initial program is an interactive shell, but an alternate program can be supplied on the ``flux start`` command line. Once the initial program terminates, the Flux session is considered complete and brokers exit.
```
Comment (grondo) [inline quickstart.rst:185]: flux-core docs uses `wireup` (it is added to our dictionary anyway). Up to you but maybe just add this one to the spell.en.pws here?

**Comment by SteVwonder** [comment]
Thanks @gonsie for the quick review.    Looks like mergify error'd out due to a permissions issue??  If you are willing to push the button, we can figure that out on the next PR.  If not, I'll poke at it on Monday.  I'm officially signing off for the day. :)  Have a good weekend!

**Comment by grondo** [comment]
Hm, yeah mergify is getting the same error I saw last week in flux-core. I'll forcibly merge.

**Pull request merged**